odoo/_orm/fields.py

```diff
@@ -0,0 +1,1465 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+""" High-level objects for fields. """
+from __future__ import annotations
+
+import itertools
+import logging
+import typing
+import warnings
+from operator import attrgetter
+
+from psycopg2.extras import Json as PsycopgJson
+
+from odoo.exceptions import AccessError, MissingError
+from odoo.osv import expression
+from odoo.tools import SQL, lazy_property, sql
+from odoo.tools.misc import SENTINEL, Sentinel
+
+from .utils import expand_ids, PREFETCH_MAX
+
+# we may not import models here to avoid cyclic dependencies
+# inject BaseModel here for performance reasons when import odoo._orm.model (ugly)
+if typing.TYPE_CHECKING:
+    from .models import BaseModel
+    _Fields_BaseModel = BaseModel
+else:
+    _Fields_BaseModel: type
```

Comment:
You actually don't need the `TYPE_CHECKING` case.  You could do the `else` clause all the time in this file.

Reply:
Removed.